app/core/interfaces/checkboxes/interface.js

```diff
@@ -15,50 +16,91 @@ define(['core/UIView', 'underscore'], function (UIView, _) {
   return UIView.extend({
     template: 'checkboxes/input',
     events: {
-      'change input[type=checkbox]': 'updateValue'
+      'change input[type=checkbox]': 'update',
+      'change #customText': 'update'
     },
-    updateValue: function () {
-      var values = this.$('input[type=checkbox]:checked');
+    delimiterize: function (out) {
+
       var delimiter = this.options.settings.get('delimiter');
+
+      if (this.options.settings.get('wrap_with_delimiter')) {
+        out = delimiter + out.join(delimiter) + delimiter;
+      } else {
+        out = out.join(delimiter);
+      }
+
+      return out;
+    },
+    value: function () {
+      var checkedValues = this.$('input[type=checkbox]:checked');
+
       var out = [];
 
-      if (values.length > 0) {
-        for (var i = 0; i < values.length; i++) {
-          out.push(values[i].value);
-        }
+      if (checkedValues.length > 0) {
 
-        // Wrap the values with a pair of delimiters to allow strict searching for a single value
-        if (this.options.settings.get('wrap_with_delimiter')) {
-          out = delimiter + out.join(delimiter) + delimiter;
-        } else {
-          out = out.join(delimiter);
+        for (var i = 0; i < checkedValues.length; i++) {
+          out.push(checkedValues[i].value);
         }
+
+        return out
+
       } else {
+
         out = '';
+
+        return out
       }
 
-      this.$('input#js-checkbox-values').val(out);
+
+    },
+    customValue: function () {
+
+      var customValues = this.$('#customText')[0].value;
+
+      return customValues;
+    },
+
+    update: function () {
+      var out = []
+
+      out.push(this.value())
+
+      out.push(this.customValue())
```

Comment:
We should check if the value exists before pushing it into the list, to prevent empty elements.